qcelemental/models/molecule.py

```diff
@@ -438,13 +456,26 @@ def from_data(cls, data, dtype, orient=False, **kwargs):
         Molecule
             A constructed molecule class.
         """
+        if dtype is None:
+            if isinstance(data, str):
+                dtype = "string"
+            elif isinstance(dat, np.array):
+                dtype = "numpy"
+            elif isinstance(data, dict):
+                dtype = "dict"
+            else:
+                raise TypeError("Input type not understood, please supply the 'dtype' kwarg.")
 
-        if dtype == "psi4":
-            input_dict = cls._molecule_from_string_psi4(data)
+        if dtype in ["string", "psi4"]:
+            input_dict = to_schema(from_string(data)["qm"], 1)["molecule"]
         elif dtype == "numpy":
-            input_dict = cls._molecule_from_numpy(data,
-                                                  frags=kwargs.pop("frags", []),
-                                                  units=kwargs.pop("units", "angstrom"))
+            data = {
+                "geom": data[:, 1:],
+                "elez": data[:, 0],
+                "units": kwargs.pop("units", "Angstrom"),
+                "fragment_separators": kwargs.pop("frags", [])
```

Comment:
your `frags` definitely the [same as mine](https://github.com/MolSSI/QCElemental/blob/master/qcelemental/molparse/from_arrays.py#L249-L250)?

Reading to end, answer is yes. it's one of your internal flags, not qcschema.